webapp/src/monaco.tsx

```diff
@@ -1038,6 +1076,10 @@ export class Editor extends toolboxeditor.ToolboxEditor {
             // Reload the header if a change was made to the config file: pxt.json
             return this.parent.reloadHeaderAsync();
         }
+        if (this.editAmendmentsListener) {
+            this.editAmendmentsListener.dispose();
```

Comment:
do you need to ``this.editAmendmentsListener = undefined`` ?

Reply:
> EditAmendmentInstance

It's an EditAmendment but with a concrete range in the editor.
An EditAmendment is independent of the current state of the editor.
An EditAmendmentInstance is only valid for a specific state of the editor.